local_publisher.py

```diff
@@ -28,10 +30,13 @@ def configure(self, context):
         super().configure(context)
         self._publisher = NioPublisher(
             topic="{}.{}".format(self.local_identifier(), self.topic()))
-        self._publisher.open()
+        self._publisher.open(on_connected=self.conn_on_connected,
+                             on_disconnected=self.conn_on_disconnected)
```

Comment:
Will this callback get called when the publisher is closed? That would send the block into a warning state right before stopping the service, which we would want to avoid.

Reply:
Maybe we need a `self._is_stopping` flag that would ignore the on-disco callback if we're stopping